core: show xpubs in GetAddress for multisig

prusnak opened the pull request:
what $SUBJ says

`core mypy test` fails with the following:

```
src/apps/common/layout.py:80: error: Argument 1 to "Paginated" has incompatible type "List[Text]"; expected "List[Component]"  [arg-type]
```

but I am not sure whether this is my fault (code works) or typing fault. Please advise.

tsusanka commented:
You have to declare `pages` as `List[ui.Component]` because mypy sees it as `List[Text]` otherwise, because something something [variance](https://mypy.readthedocs.io/en/latest/common_issues.html#variance). I'll leave the proper explanation to @matejcik :).

I've pushed the fix myself.

tsusanka commented:
I've tried to test this and using ` INTERACT=1 pytest tests/device_tests/test_msg_getaddress.py  -k "test_multisig and not test_multisig_missing"` I'm getting some weird screens.

1. The header is 'Multisig 2 of 0', which does not seem correct, but it happens on master as well.
1. Clicking on XPUBs leads back to the address screen. Maybe this is some corner case?

![Peek 2019-10-23 20-18](https://user-images.githubusercontent.com/1835345/67422230-5042c080-f5d2-11e9-8fab-0ef8b3bc2a0f.gif)

prusnak commented:
This is really weird, but as you said, this was already present in master. Investigating.

prusnak commented:
Great catch! Fixed and squashed the changes into a single commit.

We were not handling `MultisigRedeemScriptType.nodes` correctly in `GetAddress`.

When this is set we should use the value from this field rather than from the `pubkeys` field.

This is fixed here:

https://github.com/trezor/trezor-firmware/pull/645/commits/2549614fd8ee29f5dd81a9c4a1fd764a04ddbc7c#diff-a8c9d634814d4d0310299a388b36dca2R62-R65

tsusanka commented on core/src/apps/wallet/get_address.py:
```diff
@@ -1,11 +1,37 @@
+async def show_xpubs(
+    ctx: wire.Context, coin: CoinInfo, pubnodes: List[HDNodeType]
+) -> bool:
+    for i, x in enumerate(pubnodes):
```
I'm not very fond of these short variable names. `i` is fine, but I'd prefer calling `x` `pubnode` or similar.

matejcik commented on core/src/apps/wallet/get_address.py:
```diff
@@ -1,11 +1,37 @@
+) -> bool:
+    for i, x in enumerate(pubnodes):
+        cancel = "Next" if i < len(pubnodes) - 1 else "Address"
+        node = bip32.HDNode(
```
hm, no, because `MessageType` in core does not support this usage

matejcik commented:
LGTM, but didn't we want to have a dedicated third button for xpubs?
(maybe leave that for a subsequent issue though?)

prusnak commented:
> LGTM, but didn't we want to have a dedicated third button for xpubs?

I want to replicate this feature to T1 later and I'd prefer consistency for now. (We have only two buttons on T1).

prusnak commented:
Merged as 9a3d8ef2c42aaaa82c048a105033ea49d0731794

prusnak commented:
The bug tsusanka found was also present in T1 code. Fixed in c145db7dd2b56a0f800422656d2eddc7f3e4a859
